### PR TITLE
Build using CMake

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

File: Dockerfile.QA [hunk 1]
```diff
@@ -41,52 +55,54 @@ FROM ${BUILD_IMAGE} AS build
...
-    mkdir -p qa/custom_models/custom_nobatch_float32_float32_float32/1 && \
-    cp /opt/tensorrtserver/custom/libaddsub.so \
-       qa/custom_models/custom_nobatch_float32_float32_float32/1/. && \
-    mkdir -p qa/custom_models/custom_sequence_int32/1 && \
-    cp /opt/tensorrtserver/custom/libsequence.so \
-       qa/custom_models/custom_sequence_int32/1/.
+    cp -r docs/examples/model_repository/simple_string qa/L0_simple_string_example/models/.
+#    mkdir qa/L0_simple_inprocess/models && \
+#    cp /opt/tensorrtserver/bin/simple_inprocess qa/L0_simple_inprocess/. && \
+#    cp -r docs/examples/model_repository/simple qa/L0_simple_inprocess/models/.
```
[the remark this answers is not in the document]
Comment (deadeyegoodwin) [reply Dockerfile.QA]: I need to fix the in-process build. When I do I'll uncomment.

File: docs/client.rst [hunk 2]
```diff
@@ -98,22 +100,43 @@ more information about running the inference server)::
...
 to be built on additional platforms.
 
+To build the libaries and examples using CMake, first change
+directory to the root of the repo and checkout the release version of
+the branch that you want to build (or the master branch if you want to
+build the under-development version). The branch you use for the
+client build should match the version of the inference server you are
+using::
+
+  $ git checkout r19.04
```
[the remark this answers is not in the document]
Comment (deadeyegoodwin) [reply docs/client.rst]: I need to go over all the build documentation so I'll update that when I do.

**Comment by cloudhan** [comment]
I would suggest to move to modern target based CMake with
```CMake
target_include_directories(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
target_compile_definitions(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
target_compile_options(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
target_compile_features(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
target_sources(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
target_link_libraries(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
```

or even more modern cmake with additional

```CMake
target_link_options(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
target_link_directories(<target_name> <PUBLIC|PRIVATE|INTERFACE> [...])
```

**Comment by deadeyegoodwin** [comment]
@cloudhan We can't go too modern since we want to work with cmake 3.5.2 (ships with Ubuntu 16.04). But please submit pull requests for specific improvements that are compatible with that version.

**Pull request merged**